segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableKey.java

```diff
@@ -96,15 +95,14 @@ public String toString() {
 
     @Override
     public int hashCode() {
-        return HashedArray.hashCode(this.key);
+        return this.key.hashCode();
```

Comment:
`HashedArray.hashCode` is a static method so it never did cache the value. I don't think this class ever cached it.

What you probably remember was that the `HashedArray` object itself was caching the hash value - which was incorrect since the array it wraps may mutate so the hash would change.